packages/gatsby-source-wordpress/src/steps/source-nodes/update-nodes/wp-actions/update.js

```diff
@@ -304,13 +304,17 @@ const wpActionUPDATE = async ({ helpers, wpAction }) => {
   const existingNode = await getNode(nodeId)
 
   if (wpAction.referencedNodeStatus !== `publish`) {
-    // if the post status isn't publish anymore, we need to remove the node
-    // by removing it from cached nodes so it's garbage collected by Gatsby
-    const validNodeIds = cachedNodeIds.filter(cachedId => cachedId !== nodeId)
+    if (needToTouchNodes) {
+      // if the post status isn't publish anymore, we need to remove the node
+      // by removing it from cached nodes so it's garbage collected by Gatsby
+      const validNodeIds = cachedNodeIds.filter(cachedId => cachedId !== nodeId)
 
-    await setPersistentCache({ key: CREATED_NODE_IDS, value: validNodeIds })
+      await setPersistentCache({ key: CREATED_NODE_IDS, value: validNodeIds })
+    }
 
     if (existingNode) {
+      // calling touchNode here ensures owners is populated before we delete.
+      // In some cases calling delete node without touching the node first throws an error. this is a bug in Gatsby that needs to be fixed
```

Comment:
with persisted type owner tracking I do think we fixed the problem, but also we need the code still for gatsby versions before that fix - so mostly it's about comment text not being right anymore after https://github.com/gatsbyjs/gatsby/pull/37782 I think